GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -0,0 +1,120 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common.Maintenance
+{
+    public class LooseObjectsStep : GitMaintenanceStep
+    {
+        public const string LooseObjects = "loose-objects.history";
+        private readonly string looseObjectsHistoryPath = null;
+        private readonly bool ignoreTimeRestriction = false;
+
+        public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool ignoreTimeRestriction)
+            : this(context, requireCacheLock)
+        {
+            this.ignoreTimeRestriction = ignoreTimeRestriction;
+        }
+
+        public LooseObjectsStep(GVFSContext context, bool requireCacheLock)
+            : base(context, gitObjects: null, requireObjectCacheLock: requireCacheLock)

```

Comment:
`null` gitObjects? That seems like it will break.

Also: why two constructiors? Shouldn't we just make `ignoreTimeRestriction` default to `false`?

Reply:
We don't use the gitObjects so I set them to null.  If we have a need for them in the future i can take them in.  Right now I'd rather make it clear this isn't being used.

I'm updating to one constructor.

Reply:
Wow. That's totally my fault. Only the `PrefetchStep` needs the `GitObjects` member, and that is causing a lot of unneeded code passing one to the constructor of every `GitMaintenanceStep`. I'm sure there is a historical reason why that was done in the base class, but it isn't needed anymore.

While you are here, could you move the `GitObjects` out of `GitMaintenanceStep` and only into `PrefetchStep`? This will change constructors in `PostFetchStep` and the unit tests. Sorry for the pain here.

Reply:
:-)  I'm happy to do the refactor here but I'd prefer to do it in a separate PR.  This one is already so big and I don't want to overload my very kind reviewers (yourself included!).

I've added an issue to track #617